pkg/compiler/codegen.go

```diff
@@ -38,6 +38,18 @@ type codegen struct {
 	// Current funcScope being converted.
 	scope *funcScope
 
+	// A mapping from label's names to their ids.
+	labels map[string]int
+
+	// A label for the statement being visited.
+	labelFor string
+	// A label for the switch statement being visited.
+	labelSwitch string
+	// A label to be used in the next statement.
+	labelToUse string
```

Comment:
`currentLabel`?

Reply:
I don't like `currentLabel` name as it can easily been misread as _label to be jumped on_ but it is rather _label name to be used for the next statement_.

Reply:
Well, `toUse` makes me wonder about when and how it is supposed to be used, thus this suggestion.

Reply:
Renamed it to `nextLabel`